### ISSUE TITLE
how to pass std map to lua?

### ISSUE THREAD

**Issue by venjin** [shybovycha/lua-bridge-sample#3]

**Comment by shybovycha** [comment]
Same way as you would pass pretty much anything else - LuaBridge supports most (if not all) of the STL collections.

Check [sample-9](https://github.com/shybovycha/lua-bridge-sample/tree/master/9-pass-map-from-cpp-to-lua)

TL;DR:

```cpp
std::map<std::string, std::string> myMap;

// ... fill out myMap

luabridge::LuaRef myLuaFunc = luabridge::getGlobal(luaState, "myLuaFunc");

myLuaFunc(myMap);
```

in Lua this would be a table (aka associative array, aka dictionary, aka map):

```lua
function myLuaFunc(myMap)
  print("MyMap" .. myMap["key"])
end
```

**Comment by venjin** [comment]
> Same way as you would pass pretty much anything else - LuaBridge supports most (if not all) of the STL collections.
> 
> Check [sample-9](https://github.com/shybovycha/lua-bridge-sample/tree/master/9-pass-map-from-cpp-to-lua)
> 
> TL;DR:
> 
> ```cpp
> std::map<std::string, std::string> myMap;
> 
> // ... fill out myMap
> 
> luabridge::LuaRef myLuaFunc = luabridge::getGlobal(luaState, "myLuaFunc");
> 
> myLuaFunc(myMap);
> ```
> 
> in Lua this would be a table (aka associative array, aka dictionary, aka map):
> 
> ```lua
> function myLuaFunc(myMap)
>   print("MyMap" .. myMap["key"])
> end
> ```

ok, thanks a lot
